File: Message/src/main/java/com/yahoo/sdvornik/message/SimpleMessage.java
package com.yahoo.sdvornik.message;

import io.netty.buffer.ByteBuf;

public class SimpleMessage extends Message {

    private final Type type;

    protected SimpleMessage(Type type) {
        this.type = type;
    }

    @Override
    public Type getType() {
        return type;
    }

    @Override
    public long getLength() {
        return Integer.BYTES;
    }

    @Override
    public void writeContent(ByteBuf buf) {
        throw new UnsupportedOperationException("Can't write content for SimpleMessage");
    }
}

File: shared/src/main/kotlin/com/yahoo/sdvornik/message/SimpleMessage.kt
package com.yahoo.sdvornik.message

import io.netty.buffer.ByteBuf

/**
 * @author Serg Dvornik <sdvornik@yahoo.com>
 */
class SimpleMessage (override val type: Message.Type) : Message() {

    override val length: Long
        get() = Integer.BYTES.toLong()

    override fun writeContent(buf: ByteBuf) {
        throw UnsupportedOperationException("Can't write content for SimpleMessage")
    }
}